payu/models/cice.py

```diff
@@ -244,8 +254,9 @@ def setup(self):
         assert(total_runtime % setup_nml['dt'] == 0)
         setup_nml['istep0'] = int(total_runtime / setup_nml['dt'])
 
-        # Force creation of a dump (restart) file at end of run
-        setup_nml['dump_last'] = True
+        if self.model_type != 'cice':  # model_type==cice5
+            # Force creation of a dump (restart) file at end of run
+            setup_nml['dump_last'] = True
```

Comment:
Would another option be to do this from within the `cice5.py` driver? I'm not sure what would be better, as that would require a separate read/write of the namelist.

Reply:
I'm not convinced we should do this in the driver. Seems like overreach and is impossible for a user to override if done at this point.

We can add something like this to QA checks for NRI released configurations. I'd even prefer making this the default in the model because it can then be overridden if necessary.

Reply:
P.S. I realise that code change would be in `cice5`, and therefore not directly related to this PR. I am also not advocating for that code change immediately, but would support it happening _eventually_

Reply:
The `cice5` driver runs this `setup` function from the `cice` Class entirely. This change to the namelist happens whilst the cice setup function is making other changes to the cice namelist. Its hard to do this in the cice5 driver because it would require either re-opening the cice nml and changing it, and writing that. Or refactoring this `setup` function, such that the cice5 setup function can do additional stuff between the namelist being opened and closed. Or adding an optional argument to the cice setup function with a dictionary of additional changes.

 Sounds like a new issue aidan :)

Reply:
By new issue - I assume you were talking about setting `dump_last` at all in payu, rather than the new if statement.

Reply:
Correct. As you said, a different issue entirely.